### DISCUSSION TITLE
Example: html string to plain text

### DISCUSSION THREAD

**Discussion by sowinski** [boisgera/pandoc#46] [General]
Hi,

can someone help me to write some code to transform a html string into a plain text string?

```
import pandoc
html = "<p>This is a test</p>"
text = pandoc.magic(html)
print(text)
>>This is a test
```
Would be nice to see more examples in the docs

**Comment by boisgera** [comment]
Hi @sowinski ! 

The following code should do the trick:

``` pycon
>>> import pandoc
>>> html = "<p>This is a test</p>"
>>> doc = pandoc.read(html, format="html")
>>> doc
Pandoc(Meta({}), [Para([Str('This'), Space(), Str('is'), Space(), Str('a'), Space(), Str('test')])])
>>> markdown = pandoc.write(doc) 
>>> markdown
'This is a test\n'
```

Basically, `read` converts from any format to the Python format (here, what the `doc` variable contains) and `write` from the Python format to any format (markdown being the default). More relevant information can be found in the API documentation [here](https://boisgera.github.io/pandoc/api/#pandoc) and it comes with examples.

> Would be nice to see more examples in the docs

I agree that the documentation is not very beginner-friendly ; there is a lot of examples [in the doc](https://boisgera.github.io/pandoc/) but most of them are rather advanced. The use case for `pandoc` (this Python library) is mostly the automation of document analysis and transform, which is in itself a rather advanced topic ; therefore I have prioritized the documentation effort towards this use case and implicitly assume that typical user already knows a lot about pandoc (the CLI tool, its document model, etc).

If you have a simpler use case (say only convert from one format to another), I'd advise you to :

  - try the [pandoc](https://pandoc.org/) command-line tool 

        $ echo "<p>This is a test</p>" | pandoc -f html -t markdown

  - If you want to use it from python, use a thin wrapper on top of this tool, such as [pypandoc](https://pypi.org/project/pypandoc/).

But if you need / want to stick with this library and need some additional information, I'll try to answer ; [Discussions](https://github.com/boisgera/pandoc/discussions) should be the appropriate place to ask for help most of the time.

Cheers,

Sébastien
